cmd/juju/controller/destroy.go

```diff
@@ -177,6 +186,50 @@ func (c *destroyCommand) Init(args []string) error {
 	return c.destroyCommandBase.Init(args)
 }
 
+// getModelNames gets slice of model names from modelData.
+func getModelNames(data []modelData) []string {
+	return transform.Slice(data, func(f modelData) string {
+		return f.Name
+	})
+}
+
+// getTotalApplicationCount gets total number of application among all models from modelData.
+func getTotalApplicationCount(data []modelData) int {
+	totalApplicationCount := 0
+	for _, mData := range data {
+		totalApplicationCount += mData.ApplicationCount
+	}
+	return totalApplicationCount
+}
+
+// getTotalMachineCount gets total number of machines among all models from modelData.
+func getTotalMachineCount(data []modelData) int {
+	totalMachineCount := 0
+	for _, mData := range data {
+		totalMachineCount += mData.HostedMachineCount
+	}
+	return totalMachineCount
+}
```

Comment:
I don't either of these functions are necessary as this data is already available to you in modelStatus.controller (so modelStatus is a bit of a bad name)

Reply:
Agreed